### ISSUE TITLE
How to debug a custom workflow step?

### ISSUE THREAD

**Issue by kevin-james-sp** [shillner/unleash-maven-plugin#114]
TL;DR - How do I hook the eclipse debugger to the perform-tycho process in order to step through a custom CDI step I am developing?

I'm trying to write a custom workflow step to run at the end of the workflow, to add the generated Eclipse plugin site to an existing site (using the compositeContent/compositeArtifacts files). I've managed to get started by copying the source from one of the existing CDI steps and modifying it, but I'd like to hook up to the Eclipse debugger (so I can stop in the middle of my method and figure out what data I have available to build my functionality against).
I've tried starting with mvnDebug but while I can hook the debugger with that, even with the `-DforkCount=0` parameter I've seen mentioned for example here:
https://maven.apache.org/surefire/maven-surefire-plugin/examples/debugging.html

I still get a forked process which requires me to detach and reattach the debugger, which then loses the connection to the process that CDI will eventually be running in.. It stops at this point:

`[INFO] 10:04:32,971  Starting release build.`
`[WARN] Maven will be executed in interactive mode, but no input stream has been configured for this MavenInvoker instance.`
`Listening for transport dt_socket at address: 8000`

**Comment by shillner** [comment]
Hi, sorry but I didn't do any maven debugging for a longer time. If I do so I always try to debug directly in Eclipse (run a Maven debug configuration with unleash:perform-tycho as its goal. Also include the unleash sources and enable workspace artifact resolution). This is the easiest way to attach your debugger to the maven process.

Sorry but there's no time for me to trie things out, ... Hope this helps a bit.

**Comment by kevin-james-sp** [comment]
After I posted this I discovered the mvn workflow step so I'm using that instead with a customized workflow. The only thing I haven't figured out yet is if it's possible to use any of the properties generated by unleash (specifically scm.tag.name) without having to read in the releaseMetadata.properties file in my sub maven process

**Comment by shillner** [comment]
If you're using the provided mvn-step then it shouldn't be possible to access the generated release metadata. But you could extend this step and inject the metadata into this step. Then you could use a system property to pass the tag name to the forked maven process if necessary.

**Comment by kevin-james-sp** [comment]
I think using properties-maven-plugin:read-project-properties is fine - I just wanted to make sure I wasn't missing something that meant I was making it harder than it needed to be..
